deltacat/tests/compactor/test_compaction_session.py

```diff
@@ -0,0 +1,406 @@
+# Allow classes to use self-referencing Type hints in Python 3.7.
+from __future__ import annotations
+import ray
+from moto import mock_s3
+import pytest
+import os
+import json
+import boto3
+from typing import Any, Dict
+from typing import List
+from boto3.resources.base import ServiceResource
+import pyarrow as pa
+
+from enum import Enum
+
+
+class PartitionKeyType(str, Enum):
+    INT = "int"
+    STRING = "string"
+    TIMESTAMP = "timestamp"
+
+
+class PartitionKey(dict):
+    @staticmethod
+    def of(key_name: str, key_type: PartitionKeyType) -> PartitionKey:
+        return PartitionKey({"keyName": key_name, "keyType": key_type.value})
+
+    @property
+    def key_name(self) -> str:
+        return self["keyName"]
+
+    @property
+    def key_type(self) -> PartitionKeyType:
+        key_type = self["keyType"]
+        return None if key_type is None else PartitionKeyType(key_type)
+
+
+MAX_RECORDS_PER_FILE: int = 1
```

Comment:
This can be moved to test constants.

Reply:
Done - moved to within `common.py` within the same package